src/Views/OperatingSystemView.vala

```diff
@@ -253,20 +250,18 @@ public class About.OperatingSystemView : Gtk.Grid {
         var continue_button = dialog.add_button (_("Restore Settings"), Gtk.ResponseType.ACCEPT);
         continue_button.get_style_context ().add_class (Gtk.STYLE_CLASS_DESTRUCTIVE_ACTION);
 
-        var result = dialog.run ();
-        dialog.destroy ();
-
-        return result == Gtk.ResponseType.ACCEPT;
-    }
-
-    private void settings_restore_clicked () {
-        if (confirm_restore_action ()) {
-            var all_schemas = get_pantheon_schemas ();
+        dialog.response.connect ((response) => {
+            if (response == Gtk.ResponseType.ACCEPT) {
+                var all_schemas = get_pantheon_schemas ();
 
-            foreach (var schema in all_schemas) {
-                reset_recursively (schema);
+                foreach (var schema in all_schemas) {
+                    reset_recursively (schema);
+                }
             }
-        }
+
+            dialog.destroy ();
```

Comment:
This is slightly different behaviour from master ?  In master the dialog is destroyed before the schemas are reset in this branch the dialog is destroyed after.  Depending on how long the reset takes the dialog could appear not to have responded (not tested as I do not want to reset my settings!)

Reply:
Fixed!